backpack/core/derivatives/shape_check.py

```diff
@@ -39,113 +52,143 @@ def check_shape(mat, like, diff=1):
 
     vec_criterion(mat, module) returns if mat is a vector.
+
+    Args:
+        mat_prod: .
+        vec_criterion: .
+
+    Returns:
+        .
     """
 
     @functools.wraps(mat_prod)
-    def wrapped_mat_prod_accept_vectors(
+    def _wrapped_mat_prod_accept_vectors(
         self, module, g_inp, g_out, mat, *args, **kwargs
     ):
         is_vec = vec_criterion(mat, module, *args, **kwargs)
-        mat_in = mat if not is_vec else add_V_dim(mat)
+        mat_in = mat if not is_vec else _add_V_dim(mat)
         mat_out = mat_prod(self, module, g_inp, g_out, mat_in, *args, **kwargs)
-        mat_out = mat_out if not is_vec else remove_V_dim(mat_out)
+        mat_out = mat_out if not is_vec else _remove_V_dim(mat_out)
 
         return mat_out
 
-    return wrapped_mat_prod_accept_vectors
+    return _wrapped_mat_prod_accept_vectors
 
 
 # vec criteria
-same_dim_as_output = functools.partial(same_dim_as, name="output")
-same_dim_as_input = functools.partial(same_dim_as, name="input0")
-same_dim_as_weight = functools.partial(same_dim_as, name="weight")
-same_dim_as_bias = functools.partial(same_dim_as, name="bias")
+same_dim_as_output = functools.partial(_same_dim_as, name="output")
+same_dim_as_input = functools.partial(_same_dim_as, name="input0")
+same_dim_as_weight = functools.partial(_same_dim_as, name="weight")
+same_dim_as_bias = functools.partial(_same_dim_as, name="bias")
 
 # decorators for handling vectors
 jac_t_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_output,
 )
 
 weight_jac_t_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_output,
 )
 bias_jac_t_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_output,
 )
 
 jac_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_input,
 )
 
 weight_jac_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_weight,
 )
 
 bias_jac_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_bias,
 )
 
 
 ###############################################################################
 #       Decorators for checking inputs and outputs of mat_prod routines       #
 ###############################################################################
-def mat_prod_check_shapes(mat_prod, in_check, out_check):
-    """Check that input and output have correct shapes."""
+def mat_prod_check_shapes(
+    mat_prod: Callable, in_check: Callable, out_check: Callable
+) -> Callable[..., Tensor]:
+    """Check that input and output have correct shapes.
+
+    Args:
+        mat_prod: .
```

Comment:
Function that applies a derivative operator to multiple vectors handed in as a matrix